tools/Shared/Microsoft.TemplateEngine.CommandUtils/NativeMethods.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.InteropServices;
+using Microsoft.Win32.SafeHandles;
+
+namespace Microsoft.TemplateEngine.CommandUtils
+{
+    internal static class NativeMethods
+    {
+        internal static class Windows
+        {
+            internal const int ProcessBasicInformation = 0;
+
+            internal enum JobObjectInfoClass : uint
+            {
+                JobObjectExtendedLimitInformation = 9,
+            }
+
+            [Flags]
+            internal enum JobObjectLimitFlags : uint
+            {
+                JobObjectLimitKillOnJobClose = 0x2000,
+            }
+
+            [DllImport("kernel32.dll", CharSet = CharSet.Unicode, SetLastError = true)]
+            internal static extern SafeWaitHandle CreateJobObjectW(IntPtr lpJobAttributes, string? lpName);
+
+            [DllImport("kernel32.dll", SetLastError = true)]
+            internal static extern bool SetInformationJobObject(IntPtr hJob, JobObjectInfoClass jobObjectInformationClass, IntPtr lpJobObjectInformation, uint cbJobObjectInformationLength);
+
+            [DllImport("kernel32.dll", SetLastError = true)]
+            internal static extern bool AssignProcessToJobObject(IntPtr hJob, IntPtr hProcess);
+
+            [DllImport("kernel32.dll", CharSet = CharSet.Auto)]
+            [DefaultDllImportSearchPaths(DllImportSearchPath.System32)]
+            internal static extern IntPtr GetCommandLine();
+
+            [DllImport("ntdll.dll", SetLastError = true)]
+            [DefaultDllImportSearchPaths(DllImportSearchPath.System32)]
+            internal static extern unsafe uint NtQueryInformationProcess(SafeProcessHandle ProcessHandle, int ProcessInformationClass, void* ProcessInformation, uint ProcessInformationLength, out uint ReturnLength);
```

Comment:
This seems to be the only reason why we need `AllowUnsafeBlocks` in all importing projects, while it's not being used.
I'd probably get rid of it and remova all `AllowUnsafeBlocks`